### ISSUE TITLE
If several resource and/or variable file have same variables, the ones in the file imported first are taken into use.

### ISSUE THREAD

**Issue by ajun007** [robotframework/robotframework#1999]
If several resource and/or variable file have same variables, the ones in the file imported first are taken into use. Actually, we need those variables have their own namespaces to distinguish each other. Maybe support "private variable" or operation like "module.variable" to distinguish variables have the same name.

**Comment by pekkaklarck** [comment]
1. Changing variable priorities is not possible because it would be massive backwards incompatible change.

2. Private variables would be nice, but I'm afraid that would be too big change to be implemented in the near future. The main problem would be that they'd require some kind of a special syntax.

3. Accessing variables defined in a certain variable/resource file using syntax like `${my_resource.Variable Name}` (or perhaps something like `${my_variable::Variable Name}` to separate this from the extended variable syntax) would probably be the easiest solution. Unfortunately the original source of variables isn't available currently, so also this is going to require quite a big changes. We already have some related changes targeted to RF 2.9 so I'll tentatively assign this to that milestone.

**Comment by blueenergy** [comment]
${my_variable::Variable Name}, is it possible to support syntax like ${SELF::Variable Name}? so it means only search variable from local resource file

**Comment by pekkaklarck** [comment]
`${SELF::Variable Name}` or some other special syntax to refer variables in the same file would be a good idea.

**Comment by pekkaklarck** [comment]
I prototyped implementing this and found out that a real implementation requires so big internal changes that we don't have time for them in RF 2.9 scope. We are going to have RF 3.0 still this year and this will be one of the critical issues in that.

**Comment by mwalle** [comment]
if this gets implemented sometime, please check if it is feasable to add support for custom names for resources. Eg.

    Import Resource path/to/resource.txt AS my_resource

Then you can use

    ${my_resource.variable name}

Because this will then solve the problem with duplicate resource filenames. See
https://groups.google.com/d/msg/robotframework-users/0Cb215YSnG4/mWpGHi5dwUwJ

**Comment by pekkaklarck** [comment]
That's a good idea. Should probably be a separate issue, but naturally this one is a precondition because otherwise such support wouldn't give any benefits. Instead of `AS` we probably should use `WITH NAME` that also libraries support. If `AS` is considered much better, we should then change libraries to use it too.

**Comment by mwalle** [comment]
no, i was confused. my intention was to use the same syntax as for the library imports. so "with name" is fine.

**Comment by jussimalinen** [comment]
No time for this in 3.0.

**Comment by pekkaklarck** [comment]
I don't have time for this in RF 3.1. Descoping.
